pkg/storage/allocator_scorer.go

```diff
@@ -26,6 +26,17 @@ import (
 )
 
 const (
+	// This is a somehow arbitrary chosen upper bound on the relative error to be
+	// used when comparing doubles for equality. The assumption that comes with it
+	// is that any sequence of operations on doubles won't produce a result with
+	// accuracy that is worse than this relative error. There is no guarantee
+	// however that this will be the case. A programmer writing code using
+	// floating point numbers will still need to be aware of the effect of the
+	// operations on the results and the possible loss of accuracy.
+	// More info https://en.wikipedia.org/wiki/Machine_epsilon
+	// https://en.wikipedia.org/wiki/Floating-point_arithmetic
+	epsilon = 1e-10
```

Comment:
We already have `kv.allocator.range_rebalance_threshold` which is supposed to prevent rebalancing unless there is a 5% difference. Why are we introducing a new magic number instead of reusing that threshold here?

How safe is this number of `1e-10`? The numbers we work with here are often 5 or 6 digits (e.g. number of ranges per node), so I'd be concerned about `1e-10` being smaller than the kinds of rounding errors we could see. I think it would be better to have a much larger cutoff (or better, something proportional to the actual values instead of a constant).